.pre-commit-config.yaml

```diff
@@ -39,7 +35,8 @@ repos:
     rev: v0.1.3
     hooks:
       - id: ruff
-        args: [--fix, --exit-non-zero-on-fix, --unsafe-fixes]
+        args: [--fix, --exit-non-zero-on-fix, --unsafe-fixes, --show-fixes]
+      - id: ruff-format
```

Comment:
Shouldn't the format be before the lint?

Reply:
The formatter is guaranteed to not add new violations (if not, it's a bug and can be reported), while the linter is not guaranteed to not mess up the format. This is the official recommendation for now, eventually there might be a more elegant solution.

Reply:
But if I run the linter, and the format; wouldn't it mean my linter reports will get reported with bad lines?

Reply:
That’s why there will be a better solution in the future. I assume it likely will have to be a way to run both at once. (But that’s a guess)

Reply:
https://x.com/henryschreiner3/status/1717232907235086710?s=46&t=LsA5Nji4tS3d8uQSfmTybw